src/pydocstyle/checker.py

```diff
@@ -150,9 +151,19 @@ def check_source(
                         len(ignore_decorators.findall(dec.name)) > 0
                         for dec in definition.decorators
                     )
+                    # Only skip checking the docstring if absent.
+                    # If the docstring is present, then check it as normal
+                    name_skip = (
+                        ignore_functions is not None
+                        and not definition.docstring
+                        and bool(ignore_functions.findall(definition.name))
```

Comment:
`search()` is probably a better option than `findall()` because it only needs to find one instance of the pattern instead of all. However, function names should short enough that the performance difference is negligible, so I'm not sure a change is needed.